homeassistant/components/gree/bridge.py

```diff
@@ -1,11 +1,77 @@
 """Helper and wrapper classes for Gree module."""
+from datetime import timedelta
+import logging
 from typing import List
 
 from greeclimate.device import Device, DeviceInfo
 from greeclimate.discovery import Discovery
-from greeclimate.exceptions import DeviceNotBoundError
+from greeclimate.exceptions import DeviceNotBoundError, DeviceTimeoutError
 
 from homeassistant import exceptions
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, MAX_ERRORS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DeviceDataUpdateCoordinator(DataUpdateCoordinator):
+    """Manages polling for state changes from the device."""
+
+    def __init__(self, hass: HomeAssistant, device: Device):
+        """Initialize the data update coordinator."""
+        DataUpdateCoordinator.__init__(
+            self,
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN}-{device.device_info.name}",
+            update_interval=timedelta(seconds=60),
+        )
+        self.device = device
+        self._error_count = 0
+
+    async def _async_update_data(self):
+        """Update the state of the device."""
+        try:
+            await self.device.update_state()
+
+            if not self.last_update_success and self._error_count:
```

Comment:
Please only wrap the line that can raise in the try... except block. Use an `else:` block if needed.